content/guides/getstarted/intro-to-ai.md

```diff
@@ -0,0 +1,544 @@
+    "settings": {
+        "voice": {
+            "model": "en_US-neural"
+        },
+        ...
+```
+
+The important part here are the clients and the modelProvider. In our case we
+want to use twitter as a client and openai as the model provider because this is
+what we also set our secrets for in the `.env` file.
+
+### Run the agent
+
+Now that the configuration is done we can run our character:
+
+```bash
+pnpm start --character="characters/sol.character.json"
+```
+
+This will start the agent and you can now interact with it on twitter. The
+character will automatically start posting tweets every few hours and when we
+interact with him it will be able to perform solana actions like minting NFTs
+for example.
+
+Here you can see the
+[character in action](https://x.com/solanadevhelper/status/1882222232656847143).
+
+You can also interact with the character locally using the following command:
+
+```bash
+pnpm start:client
+```
+
+For more information follow the
+[official eliza documentation](https://elizaos.github.io/eliza/docs/quickstart/)
+
+</Steps>
+
+Disclaimer: The Solana integration and actions are not perfect yet but they are
+in constant development and are moving fast. There will be a direct integration
+of the SendAI agent kit into Eliza soon.
+
+## Using GOAT Toolkit
+
+[GOAT](https://github.com/goat-sdk/goat) is similar to Solana Agent Kit a
+collection of many different blockchain tools that also supports different
+Solana actions and other chains. It can also be integrated into the Eliza
+framework for example and supports many different wallets, frameworks and block
+chains.
+
+## On chain program agent integration
+
+If you want to use an agents reply with a solana program you need to use an
+oracle that writes the reply data into an account that your program can read.
+You can find an example of an LLM which will transfer tokens after got questions
+answered correctly to the user.
+
+[On chain program LLM interaction](https://github.com/GabrielePicco/super-smart-contracts)
+
+This is an example of an on-chain program that integrates with an LLM (AI)
+oracle to create an interactive token-dispensing agent named "Mar1o".
+
+## Key Features
+
+1. **AI Agent Interaction**
+
+   - Users can chat with Mar1o to try to earn tokens
+   - Agent evaluates user's Solana knowledge to decide token amounts
+   - Responses are in JSON format with reply text and token amount
+
+2. **Token Management**
+   - Creates MAR1O tokens with metadata
+   - Can mint between 0-10,000 tokens based on AI response
+   - Uses PDAs for token management
+
+## Example Usage
+
+```rust
+// Initialize the AI agent with personality
+const AGENT_DESC: &str = "You are an AI agent called Mar1o which can dispense MAR1O tokens...";
+
+// User interaction flow:
+1. User sends message to agent
+2. Oracle processes with LLM
+3. Agent responds with JSON: {"reply": "...", "amount": 1000}
+4. If amount > 0, tokens are minted to user
+```
```

Comment:
Can you provide more context what this code snippet is for? Its a rust snippet, but it doesn't seem like rust code other than the const.